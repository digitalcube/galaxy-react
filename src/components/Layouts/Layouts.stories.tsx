import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Layouts, LayoutsProps } from './Layouts';

const meta: Meta = {
  title: 'Components/Layouts',
  component: Layouts,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    variant: {
      options: ['default', 'login', 'signup'],
      control: { type: 'radio' },
    },
  },
};

export default meta;

const Template: Story<LayoutsProps> = (args) => (
  <>
    <Layouts {...args} />
  </>
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  variant: 'login',
  footer: 'Footer',
};
