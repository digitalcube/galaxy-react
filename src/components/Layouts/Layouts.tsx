import React, { FC, PropsWithChildren, ReactNode, } from "react";

export type LayoutsProps = PropsWithChildren<{
    variant: 'login' | 'signup';
    footer?: ReactNode
}>
export const Layouts: FC<LayoutsProps> = ({children, variant, footer}) => {
    const classNames = [
        "d-flex",
        "flex-column",
        "min-vh-100",
    ]
    switch(variant) {
        case 'signup':
            classNames.push("shifter-dashboard-signup")
            break;
        case 'login':
            classNames.push('shifter-dashboard-login')
            break;
        default:
            break;
    }
    const className = classNames.join(' ')
    return (
        <div className={className}>	        
            <div className="mt-4 mb-4 flex-grow-1 d-flex align-items-center justify-content-center">{children}</div>
            {footer}
        </div>
    )
}
