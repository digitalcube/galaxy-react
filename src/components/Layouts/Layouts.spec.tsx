import { render } from '@testing-library/react';
import React from 'react';
import { Layouts } from './Layouts';
import { Default as LayoutsStory } from './Layouts.stories';

describe('Layouts', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(
      <LayoutsStory {...(LayoutsStory.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it.each([
    [undefined, 'shifter-dashboard'],
    ['login' as const, 'shifter-dashboard-login'],
    ['signup' as const, 'shifter-dashboard-signup'],
    ['admin' as const, 'shifter-dashboard'],
  ])(
    'variant is %p, should add the %p class name',
    (variant, expectedClassName) => {
      const { container } = render(
        <Layouts variant={variant}>Dummy content</Layouts>
      );
      expect(container.firstChild).toHaveClass(expectedClassName);
    }
  );
});
