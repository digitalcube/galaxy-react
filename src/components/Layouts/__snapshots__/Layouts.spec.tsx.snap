// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column min-vh-100 shifter-dashboard-login"
  >
    <div
      class="mt-4 mb-4 flex-grow-1 d-flex align-items-center justify-content-center"
    >
      <div
        class="mx-auto d-flex flex-column flex-md-row align-items-center justify-content-center justify-content-md-between login-column"
      />
    </div>
    Footer
  </div>
</DocumentFragment>
`;
